File: Program34.c
#include<stdio.h>
#include<stdbool.h>

bool CheckPerfect(int iNo )
{
    int iCount = 0 ;
    int iSum = 0 ;

    for(iCount = 1 ; iCount < iNo/2 ; iCount ++)
    {
        if((iNo%iCount)== 0)
        {
            iSum = iSum + iCount ;
        }
    }
    if(iSum == iNo)
    {
        return true ;
    }
    else 
    {
        return false ;
    }
}

int main()
{
    int iValue = 0 ;
    int iAns = false ;

    printf("Enter any Number : \n");
    scanf("%d",&iValue);

    iAns = CheckPerfect(iValue);

    if(iAns == true )
    {
        printf("Number is Perfect Number\n");
    }
    else 
    {
        printf("Number is not Perfect Number");
    }

    return 0 ;
}

File: Program34.py
# Input : 4 
# Output : 1 2 3 4 

def displayPattern(iValue) : 
    for i in range(1,iValue+1):
        print(i,end = " ") 

def main():
    iNo = 0  

    iNo = int(input("Enter a Number : ")) 

    displayPattern(iNo)

if __name__=="__main__":
    main()